pkg/webhook/operatingsystemconfig/rsyslog.go

```diff
@@ -0,0 +1,320 @@
+			Permissions: pointer.Int32(0744),
+			Content: extensionsv1alpha1.FileContent{
+				Inline: &extensionsv1alpha1.FileContentInline{
+					Encoding: "b64",
+					Data:     gardenerutils.EncodeBase64(processRsyslogPstatsScript.Bytes()),
+				},
+			},
+		},
+	}...)
+
+	return rsyslogFiles, nil
+}
+
+func getRsyslogValues(rsyslogRelpConfig *rsyslog.RsyslogRelpConfig, cluster *extensionscontroller.Cluster) map[string]interface{} {
+	projectName := utils.ProjectName(cluster.ObjectMeta.Name, cluster.Shoot.Name)
+
+	var reportSuspensionContinuation *string
+	if rsyslogRelpConfig.ReportSuspensionContinuation != nil {
+		if *rsyslogRelpConfig.ReportSuspensionContinuation {
+			reportSuspensionContinuation = pointer.String("on")
+		} else {
+			reportSuspensionContinuation = pointer.String("off")
+		}
+	}
+
+	filters := computeLogFilters(rsyslogRelpConfig.LoggingRules)
+
+	return map[string]interface{}{
+		"target":                       rsyslogRelpConfig.Target,
+		"port":                         rsyslogRelpConfig.Port,
+		"projectName":                  projectName,
+		"shootName":                    cluster.Shoot.Name,
+		"shootUID":                     cluster.Shoot.UID,
+		"filters":                      filters,
+		"rebindInterval":               rsyslogRelpConfig.RebindInterval,
+		"timeout":                      rsyslogRelpConfig.Timeout,
+		"resumeRetryCount":             rsyslogRelpConfig.ResumeRetryCount,
+		"reportSuspensionContinuation": reportSuspensionContinuation,
+	}
+}
+
+func getRsyslogTLSValues(rsyslogRelpConfig *rsyslog.RsyslogRelpConfig) map[string]interface{} {
+	var permittedPeers []string
+	for _, permittedPeer := range rsyslogRelpConfig.TLS.PermittedPeer {
+		permittedPeers = append(permittedPeers, strconv.Quote(permittedPeer))
+	}
+
+	var authMode string
+	if rsyslogRelpConfig.TLS.AuthMode != nil {
+		authMode = string(*rsyslogRelpConfig.TLS.AuthMode)
+	}
+
+	var tlsLib string
+	if rsyslogRelpConfig.TLS.TLSLib != nil {
+		tlsLib = string(*rsyslogRelpConfig.TLS.TLSLib)
+	}
+
+	return map[string]interface{}{
+		"caPath":        rsyslogCaPath,
+		"certPath":      rsyslogCertPath,
+		"keyPath":       rsyslogKeyPath,
+		"enabled":       rsyslogRelpConfig.TLS.Enabled,
+		"permittedPeer": strings.Join(permittedPeers, ","),
+		"authMode":      authMode,
+		"tlsLib":        tlsLib,
+	}
+}
+
+func getRsyslogTLSFiles(ctx context.Context, c client.Client, cluster *extensionscontroller.Cluster, secretRefName, namespace string) ([]extensionsv1alpha1.File, error) {
+	refSecretName, err := lookupReferencedSecret(cluster, secretRefName)
+	if err != nil {
+		return nil, err
+	}
+
+	refSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      refSecretName,
+			Namespace: namespace,
+		},
+	}
+	if err = c.Get(ctx, client.ObjectKeyFromObject(refSecret), refSecret); err != nil {
+		return nil, err
+	}
+
+	if err := utils.ValidateRsyslogRelpSecret(refSecret); err != nil {
+		return nil, err
+	}
```

Comment:
As discussed, lets keep the validation for now, since there is no validation if the referenced secret is updated in the `shoot-rsyslog-relp-admission`. Instead I've added a `TODO` to remove the validation once the referenced secret is forced to be immutable and also a subtask for this in #2